### PR TITLE
fix interface mapping for xrd based on recent xrd changes

### PR DESCRIPTION
**Pull request by mojiiba**
This pull changes the interface mapping of cisco xrd node based on the recent changes of the xrd container.

### CODE REVIEW

File: topo/node/cisco/cisco.go [hunk 1]
```diff
@@ -315,31 +369,16 @@ func defaults(pb *tpb.Node) (*tpb.Node, error) {
...
-		pb.Config.Env = map[string]string{
-			"XR_INTERFACES":                  interfaceMap,
-			"XR_CHECKSUM_OFFLOAD_COUNTERACT": interfaceList,
-			"XR_EVERY_BOOT_CONFIG":           filepath.Join(pb.Config.ConfigPath, pb.Config.ConfigFile),
+	if pb.Model == "xrd" {
+		err := setXRDEnv(pb)
+		if err != nil {
+			return nil, err
 		}
 	} else {
```
Comment (alexmasi) [inline topo/node/cisco/cisco.go:377]: Should there be an error is pb.Model is unexpected (neither "xrd" or "e8000"),

if so use a switch here:

switch pb.Model {
default:
  return nil, fmt.Errorf("unexpected model %q", pb.Model)
case "xrd":
  if err := setXRDEnv(pb); err != nil {
    return nil, err
  }
case "e8000":
  if err := setE8000Env(pb); err != nil {
    return nil, err
  }
}

either way to else is unnecessary and the err checks can be merged inline

File: topo/node/cisco/cisco.go [hunk 2]
```diff
@@ -198,18 +198,92 @@ func fmtInt400(eid int) string {
...
+			pb.Config.Env["XR_EVERY_BOOT_CONFIG"] = filepath.Join(pb.Config.ConfigPath, pb.Config.ConfigFile)
+		}
+	}
+	pb.Config.Env["XR_MGMT_INTERFACES"] = "linux:eth0,xr_name=MgmtEth0/RP0/CPU0/0,chksum,snoop_v4,snoop_v6"
+	return nil
+
+}
+
+func getCiscoInterfaceID(pb *tpb.Node, eth string) (string, error) {
+	ethWithIDRegx := regexp.MustCompile(`e(t(h(e(r(n(e(t)*)*)*)*)*)*)\d+`) // check for e|et|eth|....
```
Comment (alexmasi) [inline topo/node/cisco/cisco.go:276]: Duplicate comment here?

File: topo/node/cisco/cisco.go [hunk 3]
```diff
@@ -198,18 +198,92 @@ func fmtInt400(eid int) string {
...
 
-func getCiscoInterfaceId(pb *tpb.Node, eth string) (string, error) {
-	ethWithIdRegx := regexp.MustCompile(`e(t(h(e(r(n(e(t)*)*)*)*)*)*)\d+`) // check for e|et|eth|....
+func setE8000Env(pb *tpb.Node) error {
+	interfaceList := "MgmtEther0/RP0/CPU0/0"
+	interfaceMap := "MgmtEther0/RP0/CPU0/0:eth0"
+	eths := make([]string, 0)
+	for k := range pb.Interfaces {
+		eths = append(eths, k)
+	}
```
Comment (alexmasi) [inline topo/node/cisco/cisco.go:207]: a bit simpler to declare

eths := []string{}

or

var eths []string

File: topo/node/cisco/cisco.go [hunk 4]
```diff
@@ -198,18 +198,92 @@ func fmtInt400(eid int) string {
...
+	sort.Strings(eths)
+	for _, eth := range eths {
+		ciscoInterfaceID, err := getCiscoInterfaceID(pb, eth)
+		if err == nil {
```
Comment (alexmasi) [inline topo/node/cisco/cisco.go:211]: more common pattern is to check

if err != nil {
  return err
}
interfaceList = ...

File: topo/node/cisco/cisco.go [hunk 5]
```diff
@@ -198,18 +198,92 @@ func fmtInt400(eid int) string {
...
+			interfaceList = fmt.Sprintf("%s,%s", interfaceList, ciscoInterfaceID)
+			interfaceMap = fmt.Sprintf("%s,%s:%s", interfaceMap, ciscoInterfaceID, eth)
+		} else {
+			return err
+		}
+	}
+	if pb.Config.Env == nil {
```
Comment (alexmasi) [inline topo/node/cisco/cisco.go:218]: code duplication can be avoided here:

if pb.Config.Env == nil {
  pb.Config.Env = map[string]string{}
}
if pb.Config.Env["XR_INTERFACES"] == "" {
  pb.Config.Env["XR_INTERFACES"] = interfaceMap
}
if pb.Config.Env["XR_CHECKSUM_OFFLOAD_COUNTERACT"] == "" {
  pb.Config.Env["XR_CHECKSUM_OFFLOAD_COUNTERACT"] = interfaceList
}
if pb.Config.Env["XR_EVERY_BOOT_CONFIG"] == "" {
  pb.Config.Env["XR_EVERY_BOOT_CONFIG"] = filepath.Join(pb.Config.ConfigPath, pb.Config.ConfigFile)
}

File: topo/node/cisco/cisco.go [hunk 6]
```diff
@@ -198,18 +198,92 @@ func fmtInt400(eid int) string {
...
+func setXRDEnv(pb *tpb.Node) error {
+	interfaceMap := ""
+	eths := make([]string, 0)
+	for k := range pb.Interfaces {
+		eths = append(eths, k)
+	}
+	sort.Strings(eths)
+	for _, eth := range eths {
+		ciscoInterfaceID, err := getCiscoInterfaceID(pb, eth)
+		if err == nil {
```
Comment (alexmasi) [inline topo/node/cisco/cisco.go:247]: same technique as above to reduction a level of indentation

File: topo/node/cisco/cisco.go [hunk 7]
```diff
@@ -198,18 +198,92 @@ func fmtInt400(eid int) string {
...
+			if interfaceMap == "" {
+				interfaceMap = fmt.Sprintf("linux:%s,xr_name=%s", eth, ciscoInterfaceID)
+			} else {
+				interfaceMap = fmt.Sprintf("%s;linux:%s,xr_name=%s", interfaceMap, eth, ciscoInterfaceID)
+			}
+		} else {
+			return err
+		}
+	}
+	if pb.Config.Env == nil {
```
Comment (alexmasi) [inline topo/node/cisco/cisco.go:257]: same as above

File: topo/node/cisco/cisco.go (hunk 3 above)
Comment (mojiiba) [reply topo/node/cisco/cisco.go:207]: done

File: topo/node/cisco/cisco.go (hunk 4 above)
Comment (mojiiba) [reply topo/node/cisco/cisco.go]: done

File: topo/node/cisco/cisco.go (hunk 5 above)
Comment (mojiiba) [reply topo/node/cisco/cisco.go:217]: done

File: topo/node/cisco/cisco.go (hunk 6 above)
Comment (mojiiba) [reply topo/node/cisco/cisco.go]: done

File: topo/node/cisco/cisco.go (hunk 7 above)
Comment (mojiiba) [reply topo/node/cisco/cisco.go:255]: done

**Comment by mojiiba** [comment]
@alexmasi thanks for the detailed review. I have addressed your comments.

File: topo/node/cisco/cisco.go (hunk 5 above)
Comment (mojiiba) [reply topo/node/cisco/cisco.go:217]: sorry, I missed this one. Now It is fixed.

File: topo/node/cisco/cisco.go (hunk 3 above)
Comment (mojiiba) [reply topo/node/cisco/cisco.go:207]: sure, done

**Review by marcushines — APPROVED** [review]

**Pull request merged**